parquet-column/src/main/java/parquet/example/data/Group.java

```diff
@@ -72,6 +73,8 @@ public Group getGroup(String field, int index) {
     return getGroup(getType().getFieldIndex(field), index);
   }
 
+  abstract public  void add(int fieldIndex, Primitive value);
+
```

Comment:
I added support for dictionaries to SimplePrimitiveConverter in order to aid testing. This change meant that  I could do the below. The changes to Simple... are not essential for the dictionary support but seemed like a good way to test

  public void addValueFromDictionary(int dictionaryId) {
    parent.getCurrentRecord().add(index, dictionaryPrimitives[dictionaryId]);
  }